### PR TITLE
Refs #15902 -- Made set_language() view always set the current language in a cookie

### PR DESCRIPTION
**Pull request by claudep**
The plan is to later deprecate/remove storing the language in the session.

### CODE REVIEW

**Comment by ziima** [comment]
I'm confused by the deprecation pathway. Based on comments in the previous PR #7117 I'd expect storing the language in session should be marked as deprecated.

**Comment by claudep** [comment]
I'm not sure, @timgraham, should we start deprecation in 2.1 and remove in 3.0, or start in 2.2 and remove in 3.1? Does it make a difference wrt LTS to LTS upgrade path?

File: tests/view_tests/tests/test_i18n.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def test_setlang(self):
         response = self.client.post('/i18n/setlang/', post_data, HTTP_REFERER='/i_should_not_be_used/')
         self.assertRedirects(response, '/')
         self.assertEqual(self.client.session[LANGUAGE_SESSION_KEY], lang_code)
+        # The code also sets the language in a cookie (see #15902 for the plan)
```
Comment (timgraham) [inline tests/view_tests/tests/test_i18n.py]: I would write this comment so it doens't have to be rewritten later: `# The language is set in a cookie.`

**Comment by claudep** [comment]
Thanks for finishing the work!

**Pull request merged**